### ISSUE TITLE
Views extracting YAML fields with line continuation characters fail

### ISSUE THREAD

**Issue by jackwhelpton** [jdkato/txtlint#1018]
### Check for existing issues

- [x] Completed

### Environment

OS: Linux
Install method: direct download
Vale version: 3.12.0

### Describe the bug / provide steps to reproduce it

YAML file such as:

```yaml
openapi: 3.0.1
info:
  description: "Line 1\
    \ Line 2"
```

view:

```yaml
engine: dasel
scopes:
  # API info
  - expr: info.description
    name: title
```

Error:

```
E100 [test.yaml] Runtime error

'Line 1 Line 2' not found

Execution stopped with code 1.
```

Verified this extracts the correct string from the Dasel Playground.

**Comment by jdkato** [comment]
This will be fixed in the next release.

**Comment by nateKlaux** [comment]
@jdkato I'm still seeing this issue, but oddly intermittently. It seems to be related to description length, though. Longer multiline content that's 300+ chars after folding triggers E100 error, while shorter content seems to pass more reliably. 

From my CLI: 

```bash
$ for i in {1..10}; do result=$(vale test.yaml 2>&1); if echo "$result" | grep -q "E100"; then echo "Run $i: FAILED"; else echo "Run $i: OK"; fi; done

Run 1: FAILED
Run 2: OK
Run 3: FAILED
Run 4: OK
Run 5: FAILED
Run 6: FAILED
Run 7: FAILED
Run 8: OK
Run 9: FAILED
Run 10: FAILED
```

Test files to repro:
[openapi-prose.yml](https://github.com/user-attachments/files/24537655/openapi-prose.yml)
[test.yaml](https://github.com/user-attachments/files/24537654/test.yaml)

```
StylesPath = styles
MinAlertLevel = suggestion

[*.yaml]
BasedOnStyles = Vale
View = openapi-prose
```

**Comment by nateKlaux** [comment]
@jdkato I'm pretty sure trailing spaces on content lines inside folded (`>`) block scalars cause this. When Dasel folds the text, a trailing space + the fold-to-space produces a double-space in the extracted value that doesn't match the source. Vale's text-location step then fails with `'...' not found`.

The failure is non-deterministic and scales with file size (more multiline descriptions = higher chance of E100).

**Minimal repro:**

```yaml
# repro.yaml
openapi: 3.0.0
info:
  title: Test
  version: '1.0'
paths:
  /a:
    get:
      description: >
        First line with trailing space.␣
        Second line continues here.
```

Note the trailing space (shown as `␣`) after `space.`. With ~35+ such blocks in a file, E100 becomes likely.

**Workaround:** strip trailing whitespace before linting. My file had 2,802 lines with trailing whitespace across 212 description blocks. After stripping: 0 E100s across 5 consecutive runs.

Vale Version:  Vale 3.14.0

**Comment by osfameron** [comment]
We are also seeing this issue.
Cleaning up the YAML to make it more "regular" (e.g. removing repeated spaces etc.) does seem to help.

(I think this means this is a bug, of course: there is no requirement for YAML to be pretty and regular, as long as it's *valid*...

In any case, it seems likely that the issue is upstream with `dasel`?)

**Comment by jdkato** [comment]
It’s probably on our end, to be honest. I’ll look into it — planning on getting a new release out in the next few days.
